openfl/interface/interactive_api/experiment.py

```diff
@@ -183,7 +183,10 @@ def _export_python_env():
         requirements_generator = freeze.freeze()
 
         def is_package_has_version(package: str) -> bool:
-            return '==' in package and package != 'pkg-resources==0.0.0'
+            return ('==' in package
+                    and package not in ['pkg-resources==0.0.0', 'pkg_resources==0.0.0']
+                    and '-e ' not in package
+                    )
```

Comment:
I would recommend using a regular expression that will strictly match `package==version` template:

`([a-z-_]+(?<!pkg-resources|pkg_resources))==((\d!)?\d(.\d)*((a|b|rc)\d)?(.post\d)?(.dev\d)?)`

Although it is a lot more confusing

Reply:
In this case, there is no need to speed up the code, therefore more readable code is preferred. A regular expression is harder to understand and change.

Reply:
I also vote against regexs 😁